core/src/main/java/org/elasticsearch/ingest/IngestServicesHolder.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.ingest;
+
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.script.ScriptService;
+
+
+public class IngestServicesHolder {
+
+    private final Environment env;
+    private final ScriptService scriptService;
+    private final TemplateService templateService;
+    private final ThreadContext threadContext;
+
+    public IngestServicesHolder(Environment env, ScriptService scriptService, TemplateService templateService,
+                                ThreadContext threadContext) {
+        this.env = env;
+        this.scriptService = scriptService;
+        this.templateService = templateService;
+        this.threadContext = threadContext;
+    }
+
+    public Environment getEnv() {
```

Comment:
I also personally don't think getters help with anything when this is a simple struct. I would just make all the members public since they are final.